src/openms/source/ANALYSIS/OPENSWATH/MRMIonSeries.cpp

```diff
@@ -257,10 +257,24 @@ namespace OpenMS
 
   void MRMIonSeries::annotateTransition(ReactionMonitoringTransition& tr, const TargetedExperiment::Peptide peptide, const double precursor_mz_threshold, double product_mz_threshold, bool enable_reannotation, std::vector<String> fragment_types, std::vector<size_t> fragment_charges, bool enable_specific_losses, bool enable_unspecific_losses, int round_decPow)
   {
-    CVTermList interpretation;
+    OPENMS_PRECONDITION(peptide.hasCharge(), "Cannot annotate transition without a peptide charge state")
+    // TODO: we should not have transitions without charge states
+    // OPENMS_PRECONDITION(tr.isProductChargeStateSet(), "Cannot annotate transition without a charge state")
```

Comment:
In some cases, we don't know the product charge state, e.g. if the transition was annotated differently before. In these cases (and if the fragment charge is -1), it will either be reannotated or flaged as unannotated. Exactly, `fragment_charges` contains the "allowed" fragment charge states.